Sources/SwiftOperatorPrecedence/OperatorPrecedenceError.swift

```diff
@@ -0,0 +1,45 @@
+//===------------------ OperatorPrecedenceError.swift ---------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+import SwiftSyntax
+
+/// Describes errors that can occur when working with operator precedence graphs.
+public enum OperatorPrecedenceError: Error {
+  /// Error produced when a given precedence group already exists in the
+  /// precedence graph.
+  case groupAlreadyExists(existing: PrecedenceGroup, new: PrecedenceGroup)
+
+  /// The named precedence group is missing from the precedence graph.
+  case missingGroup(PrecedenceGroupName, referencedFrom: SyntaxProtocol?)
```

Comment:
The general paradigm is that we want to avoid `SyntaxProtocol` wherever possible to avoid the cost of existentials, so this should be `Syntax`. The only places where I’m usually using `SyntaxProtocol` as a type that’s not a generic requirement are test helpers that are not performance critical at all.

Reply:
Okay!